### PR TITLE
XS✔ ◾ Modular Monolith Testing Rule

### PR DESCRIPTION
**Pull request by danielmackay**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

https://github.com/SSWConsulting/SSW.SAW.General/issues/2

> 2. What was changed?

This pull request introduces a new rule for testing strategies in modular monoliths, focusing on unit, integration, and workflow testing. The document outlines best practices and provides guidelines to ensure robust functionality and maintainable code across multiple modules.

Key additions include:

* **Metadata and Introduction**:
  * Added metadata for SEO and author information.
  * Provided an introduction emphasizing the importance of a strong testing strategy in modular monolith applications.

* **Unit Testing**:
  * Described the role of unit testing in validating core business rules and domain logic within individual modules.
  * Highlighted the importance of comprehensive unit tests covering typical and edge cases.

* **Integration Testing**:
  * Emphasized the value of integration testing to verify module interactions and infrastructure dependencies.
  * Suggested strategies for effective integration testing, including avoiding mocks, using real databases, and isolating test infrastructure.

* **Workflow Testing**:
  * Explained the significance of workflow tests in covering key business flows and simulating real user actions across multiple modules.
  * Advised on designing workflow tests to ensure all modules work together seamlessly. ([rules/modular-monolith-testing/rule.mdR1-R70](diffhunk://#diff-ccfe2674be7f7ed201e171cc590d9f0238ae426dc0afa5eac93d3786e4f647dbR1-R70)

> 3. Did you do pair or mob programming (list names)?

No

### CODE REVIEW

File: rules/modular-monolith-testing/rule.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* **Ensuring that each step in the workflow** completes successfully and transitions correctly to the next, providing a realistic assessment of application stability.
+
+While workflow tests are the most complex and resource-intensive, they provide confidence that major application processes are functioning correctly and as intended.
+
+::: greybox
+Design workflow tests that cover the main paths users take, ensuring all modules work together to deliver expected end results.
+:::
+::: good
+Figure: Good Example - Workflow tests simulate full user journeys, giving comprehensive validation of the application’s functionality.
+:::
```
Comment (lukecookssw) [inline rules/modular-monolith-testing/rule.md:68]: I would remove this figure, or add a good example to go along with it.

File: rules/modular-monolith-testing/rule.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+* **Verification of isolated business logic** within each module, such as validation rules or calculations.
+* **Maintenance of a clean, well-defined contract** for each module’s internal methods.
+
+::: greybox
+Ensure each module’s domain logic has comprehensive unit tests that cover typical and edge cases.
+:::
+::: good
+Figure: Good Example - Unit tests confirm the integrity of domain logic in isolation from external dependencies.
+:::
```
Comment (lukecookssw) [inline rules/modular-monolith-testing/rule.md:35]: I would remove this figure, or add a good example to go along with it.

**Review by lukecookssw — APPROVED** [review]
LGTM. Couple of small changes (either remove the figures, or add images to them)
